ui/src/dashboards/components/variablesControlBar/VariablesControlBar.tsx

```diff
@@ -0,0 +1,129 @@
+// Libraries
+import React, {PureComponent} from 'react'
+import {connect} from 'react-redux'
+import _ from 'lodash'
+
+// Components
+import VariableDropdown from 'src/dashboards/components/variablesControlBar/VariableDropdown'
+import {EmptyState, ComponentSize} from 'src/clockface'
+
+// Styles
+import 'src/dashboards/components/variablesControlBar/VariablesControlBar.scss'
+
+// Actions
+import {selectValue} from 'src/variables/actions'
+
+// Types
+import {AppState} from 'src/types/v2'
+import {VariablesState} from 'src/variables/reducers'
+
+// Decorators
+import {ErrorHandling} from 'src/shared/decorators/errors'
+
+interface OwnProps {
+  dashboardID: string
+}
+
+interface StateProps {
+  variables: VariablesState
+}
+
+interface DispatchProps {
+  selectValue: typeof selectValue
+}
+
+type Props = StateProps & DispatchProps & OwnProps
+
+@ErrorHandling
+class VariablesControlBar extends PureComponent<Props> {
+  render() {
+    // return <div className="variables-control-bar" />
+    if (_.isEmpty(this.variablesForDashboard)) {
+      return (
+        <div className="variables-control-bar">
+          <EmptyState
+            size={ComponentSize.ExtraSmall}
+            className="variables-control-bar--empty"
+          >
+            <EmptyState.Text text="To see variable controls here, use a variable in a cell query" />
+          </EmptyState>
+        </div>
+      )
+    }
+
+    return (
+      <div className="variables-control-bar">
+        {Object.keys(this.variablesForDashboard).map(variableID => {
+          const initialSelected = _.get(
+            this.valuesForDashboard,
+            `${variableID}.selectedValue`
+          )
+          const name = _.get(
```

Comment:
I commented below about selectors... I think this is also a good example of what could be a selector that gets used within the `mstp` for the `VariableDropdown`.